utils/utils_test.go

```diff
@@ -150,3 +151,32 @@ func TestCoerceInterfaceMapToStringMap(t *testing.T) {
 		})
 	}
 }
+
+func TestParseUintHex(t *testing.T) {
+	t.Parallel()
+
+	evmUint256Max, ok := (&big.Int{}).SetString("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", 0)
+	assert.True(t, ok)
+	tests := []struct {
+		name      string
+		input     string
+		want      *big.Int
+		wantError bool
+	}{
+		{"basic", "0x09", big.NewInt(9), false},
+		{"large number", "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", evmUint256Max, false},
```

Comment:
What about tests for 1) number that would be negative if signed 2) number that's too large